### PR TITLE
Create Demo_Posts_Creation_Preparation

### LINKED ISSUE #40: Create Demo_Posts_Creation_Preparation
## Description

Demo post creation preparation will ensure that the plugin being checked is provided with all the necessary posts it needs to conduct the audit.

## Acceptance Criteria

- Class `Demo_Posts_Creation_Preparation` should be created and exists at `includes/Checker/Preparations`
- Class `Demo_Posts_Creation_Preparation` should extend `Preparation` interface
- Class `Demo_Posts_Creation_Preparation` has a constructor.
    - The constructor accepts a `$posts` parameter
    - The `$posts` parameter passed will be an array of posts data. Each item in the array will contain post data compatible with the [`wp_insert_post()`](https://developer.wordpress.org/reference/functions/wp_insert_post/) function.
    - The post data passed will be assigned to a protected property to be used later.
- `prepare()` must be implemented from `Preparation` interface
    - The method will loop over the post data array passed in the constructor and create each post using the `wp_insert_post()` function.
    - For each post successfully created, the post ID returned will be stored in an array.
    - The cleanup function returned will loop over the array of post IDs created and remove each post from the database.

## Tests Coverage

- `Demo_Posts_Creation_Preparation` instance expected to be implementing of `Preparation` interface
- `prepare()` method adds the passed posts to the database.
- The cleanup function returned by `prepare()` removes the posts from the database.

### PR DESCRIPTION
**Pull request by jjgrainger**
Adds the `Demo_Posts_Creation_Preparation` class.

- Addresses acceptance criteria in #40 

Closes #40

### CODE REVIEW

File: includes/Checker/Preparations/Demo_Posts_Creation_Preparation.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ *
+ * @since n.e.x.t
+ */
+class Demo_Posts_Creation_Preparation implements Preparation {
+
+	/**
+	 * An array of posts data to create.
+	 *
+	 * @since n.e.x.t
+	 * @var arrat
```
Comment (mukeshpanchal27) [inline includes/Checker/Preparations/Demo_Posts_Creation_Preparation.php]:
```suggestion
	 * @var array
```

**Review by felixarntz — APPROVED** [review]
@jjgrainger LGTM, just the small nit-pick from @mukeshpanchal27 needs to be addressed.

File: includes/Checker/Preparations/Demo_Posts_Creation_Preparation.php (hunk 1 above)
Comment (jjgrainger) [reply includes/Checker/Preparations/Demo_Posts_Creation_Preparation.php]: @mukeshpanchal27 I've addressed the typo, please can review and approve, thanks!

**Review by mukeshpanchal27 — APPROVED** [review]
Thanks @jjgrainger, LGTM!

**Pull request merged**